src/coreclr/jit/lsraxarch.cpp

```diff
@@ -1395,7 +1395,7 @@ int LinearScan::BuildCall(GenTreeCall* call)
 
         // We could use RefTypeKill, but RefTypeFixedReg is used less commonly, so the check for delayRegFree
         // during register allocation should be cheaper in terms of TP.
-        RefPosition* pos = newRefPosition(REG_SWIFT_ERROR, currentLoc, RefTypeFixedReg, call, RBM_SWIFT_ERROR);
+        RefPosition* pos = newRefPosition(REG_SWIFT_ERROR, currentLoc + 1, RefTypeFixedReg, call, RBM_SWIFT_ERROR);
```

Comment:
We were hitting this assert: https://github.com/dotnet/runtime/blob/71f8fb65a5a28018901823c19de57fe9451ab3b1/src/coreclr/jit/lsrabuild.cpp#L533-L535

In the baseline we create these RefPositions:
```
DefList: { N047.t48. PUTARG_REG; N051.t49. PUTARG_REG }
N075 ( 20,  8) [000016] --CXG------                         ▌  CALL ind unman long   REG NA $103
<RefPosition #69  @75  RefTypeFixedReg <Reg:x0 > BB02 regmask=[x0] minReg=1 wt=100.00>
<RefPosition #70  @75  RefTypeUse <Ivl:17> BB02 regmask=[x0] minReg=1 last fixed wt=100.00>
<RefPosition #71  @75  RefTypeFixedReg <Reg:x21> BB02 regmask=[x21] minReg=1 wt=100.00>
<RefPosition #72  @75  RefTypeUse <Ivl:19> BB02 regmask=[x21] minReg=1 last fixed wt=100.00>
Last use of V09 between PUTARG and CALL. Removing occupied arg regs from preferences: [x0 x21]
<RefPosition #73  @75  RefTypeUse <Ivl:2 V09> LCL_VAR BB02 regmask=[x0-xip0 x19-x28] minReg=1 last wt=400.00>
<RefPosition #74  @76  RefTypeKill <Reg:x0 > BB02 regmask=[x0] minReg=1 wt=100.00>
<RefPosition #75  @76  RefTypeKill <Reg:x1 > BB02 regmask=[x1] minReg=1 wt=100.00>
<RefPosition #76  @76  RefTypeKill <Reg:x2 > BB02 regmask=[x2] minReg=1 wt=100.00>
<RefPosition #77  @76  RefTypeKill <Reg:x3 > BB02 regmask=[x3] minReg=1 wt=100.00>
<RefPosition #78  @76  RefTypeKill <Reg:x4 > BB02 regmask=[x4] minReg=1 wt=100.00>
<RefPosition #79  @76  RefTypeKill <Reg:x5 > BB02 regmask=[x5] minReg=1 wt=100.00>
<RefPosition #80  @76  RefTypeKill <Reg:x6 > BB02 regmask=[x6] minReg=1 wt=100.00>
<RefPosition #81  @76  RefTypeKill <Reg:x7 > BB02 regmask=[x7] minReg=1 wt=100.00>
<RefPosition #82  @76  RefTypeKill <Reg:x8 > BB02 regmask=[x8] minReg=1 wt=100.00>
<RefPosition #83  @76  RefTypeKill <Reg:x9 > BB02 regmask=[x9] minReg=1 wt=100.00>
<RefPosition #84  @76  RefTypeKill <Reg:x10> BB02 regmask=[x10] minReg=1 wt=100.00>
<RefPosition #85  @76  RefTypeKill <Reg:x11> BB02 regmask=[x11] minReg=1 wt=100.00>
<RefPosition #86  @76  RefTypeKill <Reg:x12> BB02 regmask=[x12] minReg=1 wt=100.00>
<RefPosition #87  @76  RefTypeKill <Reg:x13> BB02 regmask=[x13] minReg=1 wt=100.00>
<RefPosition #88  @76  RefTypeKill <Reg:x14> BB02 regmask=[x14] minReg=1 wt=100.00>
<RefPosition #89  @76  RefTypeKill <Reg:x15> BB02 regmask=[x15] minReg=1 wt=100.00>
<RefPosition #90  @76  RefTypeKill <Reg:xip0> BB02 regmask=[xip0] minReg=1 wt=100.00>
<RefPosition #91  @76  RefTypeKill <Reg:xip1> BB02 regmask=[xip1] minReg=1 wt=100.00>
<RefPosition #92  @76  RefTypeKill <Reg:x21> BB02 regmask=[x21] minReg=1 wt=100.00>
<RefPosition #93  @76  RefTypeKill <Reg:lr > BB02 regmask=[lr] minReg=1 wt=100.00>
<RefPosition #94  @76  RefTypeKillGCRefs CALL BB02 regmask=[x8-xip0 x19-x28] minReg=1 wt=400.00>
Interval 22: long RefPositions {} physReg:NA Preferences=[x0-xip0 x19-x28] Aversions=[]
<RefPosition #95  @76  RefTypeFixedReg <Reg:x0 > BB02 regmask=[x0] minReg=1 wt=100.00>
<RefPosition #96  @76  RefTypeDef <Ivl:22> CALL BB02 regmask=[x0] minReg=1 fixed wt=400.00>
<RefPosition #97  @75  RefTypeFixedReg <Reg:x21> CALL BB02 regmask=[x21] minReg=1 wt=400.00>
```
Notice that `RefPosition #97` is at location 75 even though it comes after a `RefPosition` at location 76, so that on its own seems wrong (hence why I changed this). But it doesn't hit the assert because between `RefPosition #71` and `RefPosition #97` there is the kill at `RefPosition #92`.

With this change and this PR we now create these `RefPositions`:
```
DefList: { N047.t48. PUTARG_REG; N051.t49. PUTARG_REG }
N075 ( 20,  8) [000016] --CXG------                         ▌  CALL ind unman long   REG NA $103
<RefPosition #51  @75  RefTypeFixedReg <Reg:x0 > BB02 regmask=[x0] minReg=1 wt=100.00>
<RefPosition #52  @75  RefTypeUse <Ivl:17> BB02 regmask=[x0] minReg=1 last fixed wt=100.00>
<RefPosition #53  @75  RefTypeFixedReg <Reg:x21> BB02 regmask=[x21] minReg=1 wt=100.00>
<RefPosition #54  @75  RefTypeUse <Ivl:19> BB02 regmask=[x21] minReg=1 last fixed wt=100.00>
Last use of V09 between PUTARG and CALL. Removing occupied arg regs from preferences: [x0 x21]
<RefPosition #55  @75  RefTypeUse <Ivl:2 V09> LCL_VAR BB02 regmask=[x0-xip0 x19-x28] minReg=1 last wt=400.00>
<RefPosition #56  @76  RefTypeKill BB02 regmask=[x0-xip1 x21 lr] minReg=1 wt=100.00>
<RefPosition #57  @76  RefTypeKillGCRefs CALL BB02 regmask=[x8-xip0 x19-x28] minReg=1 wt=400.00>
Interval 22: long RefPositions {} physReg:NA Preferences=[x0-xip0 x19-x28] Aversions=[]
<RefPosition #58  @76  RefTypeFixedReg <Reg:x0 > BB02 regmask=[x0] minReg=1 wt=100.00>
<RefPosition #59  @76  RefTypeDef <Ivl:22> CALL BB02 regmask=[x0] minReg=1 fixed wt=400.00>
<RefPosition #60  @76  RefTypeFixedReg <Reg:x21> CALL BB02 regmask=[x21] minReg=1 wt=400.00>
```

Looking at this, however, it seems like the ref position is not inserted in the right location. We need to insert it before the definition of the call to ensure that doesn't get allocated into `x21`. In fact we should probably just mark the existing `RefPosition #53` as delay freed. I'll open a separate issue about this.